_posts/2020-12-15-why-developer-should-know-owasp.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: Why a developer should know OWASP?
+author: riikkanen
+excerpt: OWASP stands for Open Web Application Security Project. Why should I be interested in OWASP as a developer? If I'm not a hacker or cybersecurity specialist, and we don't even handle any sensitive information in our project, can I just skip the OWASP? Well, the answer is "nope".
+tags: 
+- security
+- developers
+---
+
+OWASP stands for Open Web Application Security Project. Why should I be interested in OWASP as a developer? If I'm not a hacker or cybersecurity specialist, and we don't even handle any sensitive information in our project, can I just skip the OWASP? Well, the answer is "nope".
+
+# Different projects, different requirements
+
+If you think about a big patient information system used in hospitals, it is obvious that information security must be concerned in every phase of building such a system: in planning, implementation, and testing. The system contains sensitive data, e.g., social security numbers, and personal medical reports. In addition to the valuable content, the system might interest hackers because it most likely has integrations to other critical information systems. The patient information system must have restricted access, strong authentication, audit logging, and whatnot. You might rightfully presume that the project team includes also hard-boiled information security professionals.
+
+![Information security](/img/why-owasp/hacker-1944688_640.jpg)
+
+How about a web store? It differs from a patient information system because it's by default available to everyone. However, also the web store contains information that must be protected: the customer information, accounts, credit card information, purchase history, and so on. It would also be nice if the purchase event is secure and reliable. If one thinks about big web stores like Amazon, one can predict that information security is tightly entwined in building the whole system. On the contrary, we cannot be so sure about smaller web stores. Are the building blocks of the store something that is widely used and acknowledged or has someone coded, for example, authentication by themselves?
+
+Finally, let's think about implementing a simple feedback form to an existing web site. Well, implementing a simple feedback form sounds quite an easy and straightforward task and it might look like we don't need to consider information security at all. However, a simple feedback form can provide a way for a hacker to harm other parts of the system, for example, by using an SQL injection. The form can also tempt spam bots, either filling your database (and site) with numerous false feedbacks or misusing the form to relay email spam messages to others. Furthermore, if we collect the usernames, phone numbers, and hometown, it quickly becomes a person registry, where the data must be protected.

```

Comment:
If you want to add something, then maybe some small reference to GDPR related to the person registry.

Reply:
web site -> website ?

Reply:
added GDPR and fixed typo